x/mongo/driver/connstring/connstring.go

```diff
@@ -216,7 +216,7 @@ func (p *parser) parse(original string) error {
 			p.PasswordSet = true
 		}
 
-		if len(username) > 1 {
+		if len(username) > 0 {
```

Comment:
We can get rid of this conditional altogether. The `strings.Contains` call will handle an empty string.